logging/src/main/java/org/jboss/as/logging/formatters/StructuredFormatterResourceDefinition.java

```diff
@@ -369,6 +369,9 @@ public void performRuntime(final OperationContext context, final ModelNode opera
             }
 
             final String name = context.getCurrentAddressValue();
+            if (name.endsWith(PatternFormatterResourceDefinition.DEFAULT_FORMATTER_SUFFIX)) {
+                throw new IllegalArgumentException("Formatter name cannot end with '" + PatternFormatterResourceDefinition.DEFAULT_FORMATTER_SUFFIX + "'");
```

Comment:
This needs to be internationalized and I think an `OperationFailedException` would be a better one to throw.